x-pack/plugins/ml/public/application/trained_models/models_management/test_models/models/inference_base.ts

```diff
@@ -58,21 +59,69 @@ export enum INPUT_TYPE {
 export abstract class InferenceBase<TInferResponse> {
   protected abstract readonly inferenceType: InferenceType;
   protected abstract readonly inferenceTypeLabel: string;
-  protected inputField: string;
   protected readonly modelInputField: string;
-  private inputText$ = new BehaviorSubject<string[]>([]);
+
+  protected inputText$ = new BehaviorSubject<string[]>([]);
+  private inputField$ = new BehaviorSubject<string>('');
   private inferenceResult$ = new BehaviorSubject<TInferResponse[] | null>(null);
   private inferenceError$ = new BehaviorSubject<MLHttpFetchError | null>(null);
   private runningState$ = new BehaviorSubject<RUNNING_STATE>(RUNNING_STATE.STOPPED);
+  private isValid$ = new BehaviorSubject<boolean>(false);
+  private pipeline$ = new BehaviorSubject<estypes.IngestPipeline>({});
+
   protected readonly info: string[] = [];
 
+  private subscriptions$: Subscription = new Subscription();
+
   constructor(
     protected readonly trainedModelsApi: ReturnType<typeof trainedModelsApiProvider>,
     protected readonly model: estypes.MlTrainedModelConfig,
     protected readonly inputType: INPUT_TYPE
   ) {
     this.modelInputField = model.input?.field_names[0] ?? DEFAULT_INPUT_FIELD;
-    this.inputField = this.modelInputField;
+    this.inputField$.next(this.modelInputField);
+  }
+
+  public destroy() {
+    this.subscriptions$.unsubscribe();
+  }
+
+  protected initialize(
+    additionalValidators?: Array<Observable<boolean>>,
+    additionalChanges?: Array<Observable<string | boolean>>
+  ) {
+    this.initializeValidators(additionalValidators);
+    this.initializePipeline(additionalChanges);
+  }
+
+  private initializeValidators(additionalValidators?: Array<Observable<boolean>>) {
+    const validators$: Array<Observable<boolean>> = [
+      this.inputText$.pipe(map((inputText) => inputText.some((t) => t !== ''))),
+      ...(additionalValidators ? additionalValidators : []),
+    ];
+
+    this.subscriptions$.add(
+      combineLatest(validators$)
+        .pipe(
+          map((validationResults) => {
+            return validationResults.every((v) => !!v);
+          })
+        )
+        .subscribe(this.isValid$)
+    );
+  }
+
+  private initializePipeline(additionalChanges?: Array<Observable<any>>) {
+    const formObservables$: Array<Observable<any>> = [
```

Comment:
```suggestion
    const formObservables$: Array<Observable<unknown>> = [
```

Reply:
Updated in https://github.com/elastic/kibana/pull/145357/commits/d70170fa0e3879cafb6097a71aa4383ca918714c